src/collection/mod.rs

```diff
@@ -9,8 +12,10 @@ use crate::{
 /// Generic collector of metrics
 #[derive(Debug)]
 pub struct Collection {
-    gauges: HashMap<String, IntGaugeVec>,
-    fgauges: HashMap<String, GaugeVec>,
+    /// Integer gauges of collection
+    pub gauges: HashMap<String, IntGaugeVec>,
```

Comment:
I'm not sure if I like gauges and their lifetime being separate fields, I'd probably introduce a new generic struct and couple gauges together with its lifetime, e.g.

```rust
struct Gauges<T> {
    value: HashMap<String, T>
    lifetime: lifetime::MetricLifetimeMap
}
```

And use this structure instead.

Reply:
then you would have to to enter gauges map and each gauge lifetimes map, it is way simpler to have global map per gauge type (int, float)